include/scratch.h

```diff
@@ -56,44 +75,42 @@ namespace diskann {
     inline std::vector<unsigned> &id_scratch() {
       return _id_scratch;
     }
-    inline float *dist_scratch() {
+    inline std::vector<float> &dist_scratch() {
       return _dist_scratch;
     }
 
-    inline T *aligned_query() {
-      return _aligned_query;
-    }
-    inline uint32_t *indices() {
-      return _indices;
-    }
-    inline float *interim_dists() {
-      return _interim_dists;
-    }
+   private:
+    uint32_t _L;
+    uint32_t _R;
+    uint32_t _maxc;
 
-    inline std::vector<float> &occlude_factor() {
-      return _occlude_factor;
-    }
+    T *_aligned_query = nullptr;
 
-    inline PQScratch<T> *pq_scratch() {
-      return _pq_scratch;
-    }
+    PQScratch<T> *_pq_scratch = nullptr;
 
-   private:
-    std::vector<Neighbor>    _pool;
-    tsl::robin_set<unsigned> _visited;
-    NeighborPriorityQueue    _best_l_nodes;
-    tsl::robin_set<unsigned> _inserted_into_pool_rs;
-    boost::dynamic_bitset<> *_inserted_into_pool_bs;
-    std::vector<unsigned>    _id_scratch;
-    float                   *_dist_scratch = nullptr;
+    // _pool stores all neighbors explored from best_L_nodes.
+    // Usually around L+R, but could be higher.
+    // Initialized to 10L, but expands as needed.
```

Comment:
I think the actual space needed is about L+R. So 10L is too much. Reducing to 3L+R. Will update the comments. The vector capacity grows with insertions anyway in case we underestimate.